vector/src/main/java/fr/gouv/tchap/activity/TchapRoomCreationActivity.java

```diff
@@ -198,16 +206,36 @@ protected void onTextChanged(CharSequence text) {
     /**
      * Process the result of the room avatar picture.
      *
-     * @param aRequestCode request ID
-     * @param aResultCode  request status code
-     * @param aData        result data
+     * @param requestCode the request id.
+     * @param resultCode  the request status code.
+     * @param data        the result data.
      */
     @Override
-    public void onActivityResult(int aRequestCode, int aResultCode, final Intent aData) {
-        super.onActivityResult(aRequestCode, aResultCode, aData);
-
-        if (REQ_CODE_UPDATE_ROOM_AVATAR == aRequestCode) {
-            onActivityResultRoomAvatarUpdate(aResultCode, aData);
+    public void onActivityResult(int requestCode, int resultCode, final Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        switch (requestCode) {
+            case REQ_CODE_UPDATE_ROOM_AVATAR:
+                onActivityResultRoomAvatarUpdate(resultCode, data);
+                break;
+            case REQ_CODE_ADD_PARTICIPANTS:
+                if (resultCode == Activity.RESULT_OK) {
+                    List<ParticipantAdapterItem> items = (List<ParticipantAdapterItem>) data.getSerializableExtra(VectorRoomInviteMembersActivity.EXTRA_OUT_SELECTED_PARTICIPANT_ITEMS);
+                    mParticipants.addAll(items);
+
+                    List<String> ids = new ArrayList<>();
+                    for(ParticipantAdapterItem item : mParticipants) {
+                        if (null != item.mUserId) {
+                            ids.add(item.mUserId);
+                        }
+                    }
+                    mRoomParams.addParticipantIds(mSession.getHomeServerConfig(), ids);
+                    createNewRoom();
+                } else if (1 == mParticipants.size()) {
+                    // the user cancels the first user selection so assume he wants to cancel the room creation.
```

Comment:
We should not cancel the room creation, the user may want to change a settings
Do nothing if the participants list is empty or limited to the user himself